cmd/admin-revoker/main.go

```diff
@@ -211,7 +240,129 @@ func (r *revoker) revokeByReg(ctx context.Context, regID int64, reasonCode revoc
 }
 
 func (r *revoker) revokeMalformedBySerial(ctx context.Context, serial string, reasonCode revocation.Reason) error {
-	return r.revokeCertificate(ctx, core.Certificate{Serial: serial}, reasonCode)
+	return r.revokeCertificate(ctx, core.Certificate{Serial: serial}, reasonCode, false)
+}
+
+// blockByPrivateKey blocks future issuance for certificates with a public key
+// matching the SPKI hash of the provided private key. Reason code must be 1
+// (Key Compromise). Keys passed to this function will be validated as a pair
+// before any actions are taken. This method does not revoke any certificates
+// directly; however 'bad-key-revoker', which references the 'blockedKeys'
+// table, will eventually revoke certificates with a matching SPKI hash.
+func (r *revoker) blockByPrivateKey(ctx context.Context, privateKey crypto.Signer, reasonCode revocation.Reason) error {
+	if reasonCode < 0 || reasonCode != 1 {
```

Comment:
```suggestion
	if reasonCode != 1 {
```

The first `if` condition was redundant with the second one.